crates/swc_ecma_minifier/src/compress/pure/strings.rs

```diff
@@ -427,21 +448,22 @@ impl Pure<'_> {
                     );
 
                     if let Some(cooked) = &mut r_first.cooked {
-                        *cooked =
-                            format!("{}{}", convert_str_value_to_tpl_cooked(&ls.value), cooked)
-                                .into()
+                        let mut str_part = convert_str_value_to_tpl_cooked(&ls.value).into_owned();
+                        str_part.reserve(cooked.len());
+                        str_part.push_str(cooked);
+                        *cooked = swc_atoms::Atom::from(str_part);
                     }
 
-                    let new: Atom = format!(
-                        "{}{}",
-                        ls.raw
-                            .clone()
-                            .map(|s| convert_str_raw_to_tpl_raw(&s[1..s.len() - 1]))
-                            .unwrap_or_else(|| convert_str_value_to_tpl_raw(&ls.value).into()),
-                        r_first.raw
-                    )
-                    .into();
-                    r_first.raw = new;
+                    let raw_str_part = ls
+                        .raw
+                        .clone()
+                        .map(|s| convert_str_raw_to_tpl_raw(&s[1..s.len() - 1]))
+                        .unwrap_or_else(|| convert_str_value_to_tpl_raw(&ls.value).into());
+
+                    let mut new_raw = String::with_capacity(raw_str_part.len() + r_first.raw.len());
```

Comment:
Same here. You don't need to allocate a new string; rather you should append to `raw_str_part`